q2_feature_table/_filter.py

```diff
@@ -80,9 +81,19 @@ def filter_features(table: biom.Table, min_frequency: int=0,
                     metadata: qiime2.Metadata=None, where: str=None,
                     exclude_ids: bool=False)\
                    -> biom.Table:
-    _filter(table=table, min_frequency=min_frequency,
-            max_frequency=max_frequency, min_nonzero=min_samples,
-            max_nonzero=max_samples, metadata=metadata,
-            where=where, axis='observation', exclude_ids=exclude_ids)
+    _filter_table(table=table, min_frequency=min_frequency,
+                  max_frequency=max_frequency, min_nonzero=min_samples,
+                  max_nonzero=max_samples, metadata=metadata,
+                  where=where, axis='observation', exclude_ids=exclude_ids)
 
     return table
+
+
+def filter_seqs(data: pd.Series, metadata: qiime2.Metadata, where: str=None,
+                exclude_ids: bool=False) -> pd.Series:
+    # Note, no need to check for missing feature IDs in the metadata, because
+    # that is basically the point of this method.
+    ids_to_keep = metadata.ids(where=where)
+    if exclude_ids is True:
+        ids_to_keep = set(data.index) - set(ids_to_keep)
+    return data[data.index.isin(ids_to_keep)]
```

Comment:
What happens when the result is completely empty (perhaps they supplied their sample metadata instead of feature ids). It seems like we may want to error if none of the IDs matched.

Reply:
No error right now: it looks like our feature table filtering doesn't error when empty (I could be wrong there, though), but distance matrix filtering does error. I am fine either way.

Reply:
I'd prefer to error (for now, at least) until https://github.com/qiime2/q2-types/issues/144 is resolved. `filter-features`/`filter-samples` should also get updated at some point, though outside the scope of this PR.

Reply:
I'm kind of thinking maybe the feature-table filters should error as well. There's nothing you can do with an empty feature table anyways, so it's just a delayed error. That's out of scope for this PR of course.

Reply:
Jinx!